### PR TITLE
Remove old timing calls from layer freezing

### PR DESCRIPTION
**Pull request by ravi-mosaicml**
Updated layer freezing to use the new timing abstraction from #146

### CODE REVIEW

**Review by Landanjs — APPROVED** [review]
LGTM! Only questions around docstrings

File: composer/algorithms/layer_freezing/layer_freezing.py [hunk 1]
```diff
@@ -26,48 +28,29 @@ def initialize_object(self) -> LayerFreezing:
...
-
-    Args:
-        current_epoch: Integer specifying the current epoch.
-        max_epochs: The max number of epochs training will run for.
-        freeze_start: The fraction of epochs to run before freezing begins.
-        freeze_level: The maximum fraction of levels to freeze.
-    """
-    # Calculate the epoch to start freezing
-    freeze_start_epoch = int(freeze_start * max_epochs)
+def _freeze_schedule(current_duration: float, freeze_start: float, freeze_level: float) -> float:
```
[the remark this answers is not in the document]
Comment (ravi-mosaicml) [reply composer/algorithms/layer_freezing/layer_freezing.py:31]: Sure, updated the docstring.

File: composer/algorithms/layer_freezing/layer_freezing.py [hunk 2]
```diff
@@ -167,8 +142,8 @@ class LayerFreezing(Algorithm):
     Runs on ``Event.EPOCH_END``.
 
     Args:
-        freeze_start: the fraction of epochs to run before freezing begins
-        freeze_level: the maximum fraction of layers to freeze
+        freeze_start (float): the fraction of training to run before freezing begins
+        freeze_level (float): the maximum fraction of layers to freeze
```
[the remark this answers is not in the document]
Comment (ravi-mosaicml) [reply composer/algorithms/layer_freezing/layer_freezing.py:146]: Neither do I :) but I added them in to be consistent. Thanks!

File: composer/algorithms/layer_freezing/layer_freezing.py [hunk 3]
```diff
@@ -184,18 +159,22 @@ def find_unused_parameters(self) -> bool:
...
+        del event  # unused
         optimizers = state.optimizers
         assert optimizers is not None
-        state.model = freeze_layers(
+        freeze_depth, freeze_percentage = freeze_layers(
             model=state.model,
             optimizers=optimizers,
-            current_epoch=state.epoch,
-            max_epochs=state.max_epochs,
+            current_duration=float(state.timer.get(state.max_duration.unit) / state.max_duration),
```
[the remark this answers is not in the document]
Comment (ravi-mosaicml) [reply composer/algorithms/layer_freezing/layer_freezing.py:173]: This calculates the amount of the training process completed, on the interval of [0,1). After writing this PR, I added a helper method to state called `state.get_elapsed_duration()` which is the same thing, but forgot to convert this one over! Thanks :)

[the remark this answers is not in the document]
Comment (Landanjs) [reply composer/algorithms/layer_freezing/layer_freezing.py]: Awesome, much cleaner!

**Comment by Landanjs** [comment]
Changes look good to me! Merge whenever you are ready

**Pull request merged**